### ISSUE TITLE
Cannot use triggersEnter and triggersExit in registry

### ISSUE THREAD

**Issue by spencern** [reactioncommerce/reaction#1098]
### Expected behavior
To be able to add functions to the `triggersEnter` and/or `triggersExit` array and have them run when visiting that route.

E.g. in the registry:

Ideal scenario - pass array of functions similar to how flowRouter currently works
```
// reaction-product-variant
  { route: '/tag/:slug?',
    name: 'tag',
    template: 'products',
    workflow: 'coreProductWorkflow',
    triggersEnter: [ function () { console.log('trigger fired'); } ] },
```

suboptimal scenario - passing a string representing a function (how to pass args?)
```
// reaction-product-variant
  { route: '/tag/:slug?',
    name: 'tag',
    template: 'products',
    workflow: 'coreProductWorkflow',
    triggersEnter: [ 'consoleLog' ] },
```

### Actual Behavior
Adding a function to the triggers enter array causes an empty array to be added to the document
e.g.
```
// reaction-product-variant
  { route: '/tag/:slug?',
    name: 'tag',
    template: 'products',
    workflow: 'coreProductWorkflow',
    triggersEnter: [] },
```
because the Schema requires an array of strings (`triggersEnter: [String]`);

Adding a String to the array, causes the following client error to be thrown and the page does not render.
```
Exception from Tracker recompute function:
debug.js:41 TypeError: Cannot read property 'length' of undefined
    at parse (index.js:85)
    at Function.compile (index.js:104)
    at Router.path (router.js:56)
    at Router.go (router.js:170)
    at router.js:68
    at withValue (dynamics_browser.js:23)
    at Router.reload (router.js:67)
    at Router.ReactionRouter.initPackageRoutes (init.js:178)
    at router.js:6
    at Tracker.Computation._compute (tracker.js:323)
```
### Steps to Reproduce the Behavior
Add function/string to `triggersEnter` property of registry in reaction package.

**Comment by prinzdezibel** [comment]
@spencern I have had the same problem. My workaround for now is to re-define the route manually again like this:

```
Meteor.startup(function(){
  ReactionRouter.group({name:"shop"}).route("/", {
    name: "index",
    action: function () {
      ReactionLayout({
        adminControlsFooter : "adminControlsFooter",
        dashboardControls : "dashboardControls",
        layout : "coreLayoutEasywine",
        layoutFooter : "footerEasywine",
        layoutHeader : "layoutHeaderEasywine",
        notFound : "productNotFound",
        template : "productLandingOnePager",
        workflow : "coreWorkflow"
      });
    },
    triggersExit : [Template.productLanding.cleanUp]
  });
```

**Comment by zenweasel** [comment]
@jshimko and I spoke about this briefly and this would probably be a client-side implementation of the `Hooks.Events`. The downside of this is that now you will have two places you need to define things for the registry. Not sure how big of a deal that is, but will probably lead to a little more confusion.

**Comment by mikemurray** [comment]
Assuming we keep our register.js files as a javascript files and not json we could do something like the following.

```
Reaction.registerPackage({
    ...
    registry: [{
      route: "/cart/checkout",
      name: "cart/checkout",
      triggersEnter: [
        // name that function!
        function checkPermissions() {
          // Do something on route enter
        }
      ]
    }]
  }
});
```

Then, when we register the routes we could register the hooks based on the function name; possibly defaulting the name if one was not provided.

```
for (let trigger of triggersEnter) {
  const name = trigger.name // Get the function name

  Hooks.Events.add(name, trigger)
}
```

Before the registry entry is saved to the database, we strip the function declarations, and replace them with their name.

Then on route enter / route leave we can trigger the hooks based on the name of the functions we just registered.

**Comment by jshimko** [comment]
Yep.  I've been digging into this and that's pretty much exactly where I was headed.

**Comment by jshimko** [comment]
This was fixed by #1253.
